core/shared/src/main/scala/zio/logging/LogGroupEquivalence.scala

```diff
@@ -0,0 +1,12 @@
+package zio.logging
+
+final case class LogGroupEquivalence[A](equivalent: (A, A) => Boolean)
```

Comment:
A data type like this naturally supports: `&&`, `||`, and `!`, all derived from Boolean.